ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-TobaccoUse.xsl

```diff
@@ -0,0 +1,182 @@
+                        </code>
+                        <valueQuantity>
+                            <xsl:call-template name="hoeveelheid-to-Quantity">
+                                <xsl:with-param name="in" select="."/>
+                            </xsl:call-template>
+                        </valueQuantity	>
+                    </component>
+                </xsl:for-each>
+                
+                <xsl:for-each select="waarneming_gebruik/pack_years">
+                    <component>
+                        <code>
+                            <coding>
+                                <system value="http://snomed.info/sct"/>
+                                <code value="401201003"/>
+                                <display value="pakjaren"/>
+                            </coding>
+                        </code>
+                        <valueInteger>
+                            <xsl:attribute name="value" select="./@value"/>
+                        </valueInteger>
+                    </component>
+                </xsl:for-each>
+            </Observation>
+        </xsl:for-each>
+    </xsl:template>
+    
+    <xd:doc>
+        <xd:desc>Template to generate a unique id to identify this instance.</xd:desc>
+    </xd:doc>
+    <xsl:template match="tabak_gebruik" mode="_generateId">
+        <xsl:variable name="parts">
+            <xsl:text>TobaccoUse</xsl:text>
+            <xsl:value-of select="tabak_gebruik_status/@displayName"/>
+            <xsl:value-of select="soort_tabak_gebruik/@displayName"/>
+            <xsl:value-of select="waarneming_gebruik/start_datum/@value"/>
+            <xsl:value-of select="waarneming_gebruik/stop_datum/@value"/>
+            <xsl:value-of select="waarneming_gebruik/hoeveelheid/@value"/>
+            <xsl:value-of select="waarneming_gebruik/pack_years/@value"/>
+            <xsl:value-of select="toelichting/@value"/>
+        </xsl:variable>
+        <xsl:value-of select="substring(replace(string-join($parts, '-'), '[^A-Za-z0-9-.]', ''), 1, 64)"/>
+    </xsl:template>
+    
+    <xd:doc>
+        <xd:desc>Template to generate a display that can be shown when referencing this instance.</xd:desc>
+    </xd:doc>
+    <xsl:template match="tabak_gebruik" mode="_generateDisplay">
+        <xsl:variable name="parts">
+            <xsl:text>TobaccoUse</xsl:text>
+            <xsl:value-of select="tabak_gebruik_status/@displayName"/>
+            <xsl:value-of select="soort_tabak_gebruik/@displayName"/>
+            <xsl:if test="waarneming_gebruik/start_datum/@value">
+                <xsl:value-of select="concat('start datum', waarneming_gebruik/start_datum/@value)"/>
+            </xsl:if>
+            <xsl:if test="waarneming_gebruik/stop_datum/@value">
+                <xsl:value-of select="concat('start datum', waarneming_gebruik/stop_datum/@value)"/>
+            </xsl:if>
+        </xsl:variable>
+        <xsl:value-of select="string-join($parts, ', ')"/>     
+    </xsl:template>
```

Comment:
Wat betreft _generateDisplay had ik met Pieter afgesproken om het qua Observations erg beperkt te houden, zodat implementers niet verleid worden om iets te bouwen op de display waarde. Zie bijv:

https://github.com/Nictiz/HL7-mappings/blob/zib2020/ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-HeadCircumference.xsl
(Alleen iets als 'Tobacco use observation' en, iets van datum en de toelichting, that's it)

Reply:
Klinkt redelijk. Ik zal wat het aanpassen. De toelichting lijkt me dan net niet iets wat er nog bij bij moet.